imgcreate/live.py

```diff
@@ -1030,14 +1030,186 @@ def _get_excluded_packages(self):
         return ["kernel.ppc"] + \
                ppcLiveImageCreator._get_excluded_packages(self)
 
+class aarch64LiveImageCreator(LiveImageCreatorBase):
+    """ImageCreator for aarch64 machines"""
+    def __init__(self, *args, **kwargs):
+        LiveImageCreatorBase.__init__(self, *args, **kwargs)
+        self._efiarch = None
+
+    def _get_xorrisofs_options(self, isodir):
+        options = []
+        if os.path.exists(os.path.join(isodir, "images/efiboot.img")):
+            options += ["-eltorito-alt-boot", "-e", "images/efiboot.img",
+                        "-no-emul-boot", "-hide-rr-moved"]
+        options += ["-rational-rock", "-joliet", "-volid", self.fslabel]
+        return options
+
+    def _get_required_packages(self):
+        return ["grub2-efi"] \
+               + LiveImageCreatorBase._get_required_packages(self)
+
+    def __copy_kernel_and_initramfs(self, isodir, version, index):
+        bootdir = self._instroot + "/boot"
+        makedirs(isodir + "/LiveOS/")
+        shutil.copyfile(bootdir + "/vmlinuz-" + version,
+                        isodir + "/LiveOS/vmlinuz" + index)
+
+        isDracut = False
+        if os.path.exists(self._instroot + "/usr/bin/dracut"):
+            isDracut = True
+
+        # FIXME: Implement a better check for how the initramfs is named...
+        if os.path.exists(bootdir + "/initramfs-" + version + ".img"):
+            shutil.copyfile(bootdir + "/initramfs-" + version + ".img",
+                            isodir + "/LiveOS/initrd" + index + ".img")
+        elif os.path.exists(bootdir + "/initrd-" + version + ".img"):
+            shutil.copyfile(bootdir + "/initrd-" + version + ".img",
+                            isodir + "/LiveOS/initrd" + index + ".img")
+        elif not self.base_on:
+            logging.error("No initramfs or initrd found for %s" % (version,))
+
+        return isDracut
+
+    def __is_default_kernel(self, kernel, kernels):
+        if len(kernels) == 1:
+            return True
+
+        if kernel == self._default_kernel:
+            return True
+
+        if kernel.startswith(b"kernel-") and kernel[7:] == self._default_kernel:
+            return True
+
+        return False
+
+    def __get_local_stanza(self, isodir):
+        return """label local
+  menu label Boot from ^local drive
+  localboot 0xffff
+"""
+
+    @property
+    def efiarch(self):
+        if not self._efiarch:
+            # for most things, we want them named boot$efiarch
+            efiarch = {"aarch64": "AA64"}
+            self._efiarch = efiarch[dnf.rpm.basearch(hawkey.detect_arch())]
+        return self._efiarch
+
+    def __copy_efi_files(self, isodir):
+        fail = False
+        files = [("/boot/efi/EFI/*/shim.efi", "/EFI/BOOT/BOOT%s.EFI" % (self.efiarch,), True),
```

Comment:
`shim.efi` is an artifact from where there wasn't support in Fedora for `ia32` and is still installed by the shim package for backward compatibility. You need to use `shimaa64.efi` instead.

I noticed that you copied this from the `x86LiveImageCreator` class, so probably needs to be changed there too.

In fact, couldn't you just use `self.efiarch` everywhere and have a base class that factors out all this common code ?